yb-voyager/src/namereg/namereg.go

```diff
@@ -375,6 +379,107 @@ func (reg *NameRegistry) LookupTableName(tableNameArg string) (sqlname.NameTuple
 	return ntup, nil
 }
 
+//================================================
+
+/*
+this function hels returning the nametuple in case the tableNameArg is only present in target or source and not both,
+so this will return a nametuple with one side populated only
+In case both the source and target not present for the table this will return error.
+*/
+
+func (reg *NameRegistry) LookupTableNameAndIgnoreOtherSideMappingIfNotFound(tableNameArg string) (sqlname.NameTuple, error) {
+	sourceName, targetName, err := reg.lookupSourceAndTargetTableNamesIgnoreIfEitherSideNotFound(tableNameArg)
+	if err != nil {
+		return sqlname.NameTuple{}, fmt.Errorf("error lookup source and target names for table [%v]: %v", tableNameArg, err)
+	}
+	ntup := NewNameTuple(reg.params.Role, sourceName, targetName)
+	return ntup, nil
+}
+func (reg *NameRegistry) lookupSourceAndTargetTableNamesIgnoreIfEitherSideNotFound(tableNameArg string) (*sqlname.ObjectName, *sqlname.ObjectName, error) {
```

Comment:
Make sure `LookupTableName` and `LookupTableNameAndIgnoreOtherSideMappingIfNotFound` use the same underlying function with different arguments (whether or not to ignore)

Reply:
so the main underlying function `lookup(...)` is the same which returns an object name (for source/target with. respective call) and an error so ignoring the error this function `lookupSourceAndTargetTableNamesIgnoreIfEitherSideNotFound ` and returning a source and target name irrespective of error

Reply:
Oh I think I got you now, I have made the respective change without duplicating the logic. having a single function returning source and target names and ignore error of lookup in either of them if `ignoreIfOtherSideMapping` is true.